app/services/geolocation_service.rb

```diff
@@ -0,0 +1,24 @@
+class GeolocationService
+  include Singleton
+
+  class << self
+    delegate :call, to: :instance
```

Comment:
love this, why did I never think of that ❤️ will steal this approach to singletons going forward